source/common/upstream/health_checker_base_impl.cc

```diff
@@ -384,6 +384,14 @@ void HealthCheckerImplBase::ActiveHealthCheckSession::onTimeoutBase() {
   handleFailure(envoy::data::core::v3::NETWORK);
 }
 
+void HealthCheckerImplBase::ActiveHealthCheckSession::start() {
+  // Start health checks only after secrets are ready for the transport socket
```

Comment:
What if you want to fetch SDS secrets from an actively health checked cluster?

Reply:
the callback is invoked in-line for all transport socket types except the TLS one. The most likely case would be that you use a raw buffer socket for the SDS cluster, health checks would start immediately in that case.

The common case would be that any SDS cluster either uses a raw buffer socket, or file based secrets which should be able to load and make progress. I could imagine a more complex setup with multiple SDS clusters that have multiple layers of secret dependencies, but at some "root" level you will need one that either doesn't use secrets or uses file based. I don't think any new challenges are created here with managing that.

Reply:
SG